packages/bsky/src/api/app/bsky/feed/getAuthorFeed.ts

```diff
@@ -82,7 +82,10 @@ export const skeleton = async (inputs: {
 
   const isFirstPageRequest = !params.cursor
   const shouldInsertPinnedPost =
-    isFirstPageRequest && params.includePins && !!actor.profile?.pinnedPost
+    isFirstPageRequest &&
+    params.includePins &&
+    !!actor.profile?.pinnedPost &&
+    actor.profile.pinnedPost.uri.includes(actor.did)
```

Comment:
Think this would be better if we either parsed out the actual DID
```ts
new AtUri(actor.profile.pinnedPost.uri).hostname
```

or explicitly checked the start of the uri 

```ts
actor.profile.pinnedPost.uri.startsWith(`at://${actor.did}`)
```

think i prefer the former

Reply:
Nice. We have the `uriToDid` util for this too 👍